### DISCUSSION TITLE
hostname issue on Solaris

### DISCUSSION THREAD

**Discussion by tropikhajma** [rabbitmq/rabbitmq-server#3116] [Other] [answered]
The hostname(1) command on Solaris does not support the '-s' and -f' flags. See https://docs.oracle.com/cd/E88353_01/html/E37839/hostname-1.html
As a result, the rabbitmq build produces several usage errors when the hostname command execution is attempted, but I am not sure if there's any impact.

Still, the attached patch fixes the issue for me on Solaris 11.4

[01-hostname.patch.txt](https://github.com/rabbitmq/rabbitmq-server/files/6669616/01-hostname.patch.txt)

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
The `.mk` changes seem fine and won't affect anyone but our development team and contributors. They are only used when nodes are built and started from source.

As for the `.ocf` file change, it is maintained by the OpenStack community. It seems reasonably safe to me.

Please submit a PR.
